src/gf.c

```diff
@@ -632,11 +634,11 @@ static void jl_cacheable_sig(
               this can be determined using a type intersection.
             */
             if (!*newparams) *newparams = jl_svec_copy(type->parameters);
-            if (i < jl_nparams(decl)) {
-                jl_value_t *declt = jl_tparam(decl, i);
+            if (decl_i) {
```

Comment:
this test can't be false (we just dereferenced `decl_i` on the previous line), nor could `jl_is_vararg_type(decl_i)` ever be true (`decl_i` was already computed as `jl_unwrap_vararg(declt)`). I think the previous test here was still correct (`jl_is_cacheable_sig` wasn't changed, and should reflect any changes here too)